meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh

```diff
@@ -309,6 +313,125 @@ function setCurrentVersion() {
+function pullSupervisor() {
+    log "Supervisor update requested through arguments..."
+    /usr/bin/resin-device-progress --percentage 25 --state "ResinOS: Preparing supervisor..."
+
+    # When requesting default supervisor update, this variable would be already filled
+    # out by the supervisor version check function. If empty, try to figure the image out.
+    if [ -z "$UPDATER_SUPERVISOR_IMAGE_NAME" ]; then
+        # Default UPDATER_SUPERVISOR_IMAGE to the one in supervisor.conf
+        if [ -z "$SUPERVISOR_REGISTRY" ]; then
+            UPDATER_SUPERVISOR_IMAGE=$SUPERVISOR_IMAGE
+        else
+            UPDATER_SUPERVISOR_IMAGE="$SUPERVISOR_REGISTRY/resin/$arch-supervisor"
+        fi
+    else
+        UPDATER_SUPERVISOR_IMAGE=$UPDATER_SUPERVISOR_IMAGE_NAME
+    fi
+
+    log "Pulling supervisor $UPDATER_SUPERVISOR_IMAGE:$UPDATER_SUPERVISOR_TAG..."
+    cachedpull "$UPDATER_SUPERVISOR_IMAGE:$UPDATER_SUPERVISOR_TAG"
+}
+
+# Update the supervisor config file with the new image and tag values.
+# If needed mount the new root filesystem as well.
+function updateSupervisorConf() {
+    local _mountnewroot="$1"
+    if [ "$_mountnewroot" == "yes" ]; then
+        local _newroot_mountpoint="/tmp/newroot"
+        local _current_root_device=$(findmnt -n --raw --evaluate --output=source /)
+        local _new_root_number
+        case $_current_root_device in
+            *2)
+                _new_root_number=3
+                ;;
+            *3)
+                _new_root_number=2
+                ;;
+            *)
+                log ERROR "Current root partition "${_current_root_device}" is not first or second root partition, aborting."
+        esac
+        local _new_root_device="${_current_root_device%?}${_new_root_number}"
+        log "Mounting new rootfs from ${_new_root_device} to ${_newroot_mountpoint}"
+        mkdir -p ${_newroot_mountpoint}
+        mount "${_new_root_device}" "${_newroot_mountpoint}"
+        SUPERVISORCONFPATH="${_newroot_mountpoint}${SUPERVISORCONF}"
+    else
+        SUPERVISORCONFPATH="$SUPERVISORCONF"
+    fi
+
+    log "Updating ${SUPERVISORCONFPATH}"
+    if grep -q "SUPERVISOR_IMAGE" "${SUPERVISORCONFPATH}"; then
+        sed --in-place "s|SUPERVISOR_IMAGE=.*|SUPERVISOR_IMAGE=$UPDATER_SUPERVISOR_IMAGE|" "${SUPERVISORCONFPATH}"
+    else
+        echo "SUPERVISOR_IMAGE=$UPDATER_SUPERVISOR_IMAGE" >> ${SUPERVISORCONFPATH}
+    fi
+    if grep -q "SUPERVISOR_TAG" "${SUPERVISORCONF}"; then
+        sed --in-place "s|SUPERVISOR_TAG=.*|SUPERVISOR_TAG=$UPDATER_SUPERVISOR_TAG|" "${SUPERVISORCONFPATH}"
+    else
+        echo "SUPERVISOR_TAG=$UPDATER_SUPERVISOR_TAG" >> ${SUPERVISORCONFPATH}
+    fi
+}
+
+# Actually updating the supervisor by tagging the new version as "latest" and updating the API
+function updateSupervisor() {
+    log "Updating supervisor tag..."
+    $DOCKER tag -f "$UPDATER_SUPERVISOR_IMAGE:$UPDATER_SUPERVISOR_TAG" "$UPDATER_SUPERVISOR_IMAGE:latest"
```

Comment:
We only need to do this tagging of the latest version on resinOS prior to 2.0.7.